packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/data_query/data_query.rs

```diff
@@ -89,43 +101,67 @@ impl From<LocalPropIdx> for PropSpecifier {
 
 /// A `DataQuery` a request for information from the document. It could be a request for a prop value,
 /// a request for a list of children, etc..
-#[derive(Debug, Clone, Default, PartialEq)]
+#[derive(Debug, Clone, Default)]
 pub enum DataQuery {
-    /// Query for all children that have a prop that matches the prescribed `PropProfile`s.
-    /// Returns the matching props
-    ChildPropProfile {
-        /// The data query will match child components that have at least one of these profiles
-        match_profiles: Vec<PropProfile>,
-    },
-
-    /// Query for all child GraphNodes,
-    /// filtering element nodes for components which contain PropProfile props given specified values
+    /// Query for all child GraphNodes, filtering element nodes based on the supplied
+    /// `filter`.
     ///
-    /// filter: a vector of `PropProfile` and `PropValue` pairings.
-    /// If a component has a prop with each specified `PropProfile`,
-    /// then it will be filtered out if any of the values don't match the specified `PropValue`.
-    /// If a component does not have a prop with one or more of the specified `PropProfiles`,
-    /// then it will be included if `include_if_missing_profile` is `true`, otherwise it will be be filtered out.
-    /// All non-component children are unaffected by the filter.
+    /// - `container`: the component whose children will be searched. For example, `PropComponent::Me`
+    /// to search your own children, or `PropComponent::Parent` to search your parent's children.
+    /// - `filter`: A composition of `ContentFilter`s. These can be combined with `Op::And`, `Op::Or`,
+    /// and `OpNot`.
     ///
-    /// For example, if the filter has one element `(PropProfile:Hidden`, `PropValue::Boolean(false))`, then
-    /// - if `include_if_missing_profile` is `true`,
-    ///   all component children that have a prop with the profile `Hidden` whose value is not `false`
-    ///   will be filtered out.
-    /// - if `include_if_missing_profile` is `false`,
-    ///   all component children that do not have a prop with the profile `Hidden` with value `false`
-    ///   will be filtered out.
-    FilteredChildren {
-        filters: Vec<DataQueryFilter>,
-        include_if_missing_profile: bool,
+    /// ## Example
+    /// ```rust
+    /// # use std::rc::Rc;
+    /// # use doenetml_core::props::{DataQuery, PropSource, ContentFilter, Op, PropProfile};
+    /// DataQuery::ComponentRefs {
+    ///    container: PropSource::Me,
+    ///    filter: Rc::new(Op::And(
+    ///      ContentFilter::IsType("section"),
+    ///      ContentFilter::HasPropMatchingProfile(PropProfile::Hidden),
+    ///    ))
+    /// }
+    /// # ;
+    /// ```
+    /// This query will return all children of the querying component that are `<section>` components
+    /// and have a prop matching the `PropProfile::Hidden` profile.
+    ComponentRefs {
```

Comment:
I'm wondering if we want parallels to the queries that return props and and queries that return `ComponentRefs`. Could the filtering/specifiying be made the same, with the only difference being that in one case, we return a `ComponentRef`, and in another case we return a `prop`?

It seems we have the cases:
1. Return a reference to possibly multiple component
2. Return a reference to a single component (could be rolled into 1)
3. Return a single prop from a single component
4. Return a single prop from multiple components
5. Return multiple props from a single component
6. Return multiple props from multiple components.

If we think of cases 1 and 2 as returns zero props, then could all follow the same structure.